target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4019-cm520-79f.dts

```diff
@@ -0,0 +1,354 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	model = "MobiPromo CM520-79F";
+	compatible = "mobipromo,cm520-79f";
+
+	aliases {
+		led-boot = &led_sys;
+		led-failsafe = &led_sys;
+		led-running = &led_sys;
+		led-upgrade = &led_sys;
+	};
+
+	chosen {
+		bootargs-append = " ubi.block=0,1 root=/dev/ubiblock0_1";
+	};
+
+	soc {
+		rng@22000 {
+			status = "okay";
+		};
+
+		mdio@90000 {
+			status = "okay";
+			pinctrl-0 = <&mdio_pins>;
+			pinctrl-names = "default";
+			reset-gpios = <&tlmm 47 GPIO_ACTIVE_LOW>;
+			reset-delay-us = <1000>;
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		tcsr@194b000 {
+			compatible = "qcom,tcsr";
+			reg = <0x194b000 0x100>;
+			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+		};
+
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+		usb2@60f8800 {
+			status = "okay";
+
+			dwc3@6000000 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				usb2_port1: port@1 {
+					reg = <1>;
+					#trigger-source-cells = <0>;
+				};
+			};
+		};
+
+		usb3@8af8800 {
+			status = "okay";
+
+			dwc3@8a00000 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				usb3_port1: port@1 {
+					reg = <1>;
+					#trigger-source-cells = <0>;
+				};
+
+				usb3_port2: port@2 {
+					reg = <2>;
+					#trigger-source-cells = <0>;
+				};
+			};
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+		};
+
+		edma@c080000 {
+			status = "okay";
+		};
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+
+		gpio-sck = <&tlmm 40 GPIO_ACTIVE_HIGH>;
```

Comment:
Oh boy, could you please mention that this will need to be updated because the binding will change?

https://www.kernel.org/doc/Documentation/devicetree/bindings/spi/spi-gpio.txt

gpio-sck -> sck-gpios
gpio-mosi -> mosi->gpios

Reply:
>  But I don't think we need this file anymore, since new devices don't get backported?!

@chunkeey AFAIK there is no clear recommendation on this topic, and frequently both files are submitted. I personally do not see much use in having these files twice, as I don't think building master with 4.14 is a common scenarios. It will just make reviewing more complicated. 
So, IMO it would be helpful to somewhat agree on only requiring 4.19 files here. I'm also not really sure what's the point of still keeping the 4.14 files around, as backporting changes shouldn't be that hard without them.

Reply:
> I'm also not really sure what's the point of still keeping the 4.14 files around, as backporting changes shouldn't be that hard without them.

Oh, from what I remember the decision to keep the old kernel around came about because of a complaint because the stable release kernel broke and the fix was non-trivial (can't find the post on the ML sadly - I think it was on the openwrt-adm though). So the kernel patches (and support) of the target should be kept around to make it possible to switch to the currenty/next stable release (19.07) for testing/developing to help out the backporters.

Reply:
Mhm, but this would not restrict us from at least having new support 4.19 only (or whatever up-to-date, unless TESTING_PATCHVER).